### ISSUE TITLE
Type inference doesn't work with a pointer of a generic unmanaged type

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#25654]
**Version Used**:  `dev15.7.x`

**Steps to Reproduce**:

```cs
class C {
    unsafe void M<T>(T* a) where T : unmanaged {
        var p = stackalloc T[10];
        M(p);
    }
}
```

**Expected Behavior**: No error

**Actual Behavior**:

`The type arguments for method 'C.M<T>(T*)' cannot be inferred from the usage. Try specifying the type arguments explicitly.`

/cc @OmarTawfik

**Comment by OmarTawfik** [comment]
Thanks @alrz for reporting. I'll take a look in the next few days.
